packages/story-editor/src/components/floatingMenu/elements/karma/shopping.karma.js

```diff
@@ -14,36 +14,44 @@
  * limitations under the License.
  */
 
+/**
+ * External dependencies
+ */
+import { waitFor } from '@testing-library/react';
 /**
  * Internal dependencies
  */
 import { Fixture } from '../../../../karma';
 import { useStory } from '../../../../app/story';
 import { tabToCanvasFocusContainer } from '../../karma/utils';
 
+function isStoryEmpty() {
+  if (document.getElementById('emptystate-message')) {
+    return true;
+  }
+  return false;
+}
```

Comment:
Can you explain what is happening here?

Reply:
Also add a comment.

Reply:
Added a comment --- 

Essentially this is a check to see if any products are on the canvas --- the `dots` don't have a `product` role or anything selectable to check based on dom elements.

For this test given a fresh story

- is the canvas empty (story empty)
- add a product --- (story not empty)
- remove the product (story empty again)

---- there's other checks for the state etc... but wanted something for the actual stage / canavs